flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/test/java/org/apache/flink/tests/util/kafka/SmokeKafkaITCase.java

```diff
@@ -0,0 +1,184 @@
+        final String outputTopic = "test-output" + "-" + UUID.randomUUID();
+
+        // create the required topics
+        final short replicationFactor = 1;
+        admin.createTopics(
+                Lists.newArrayList(
+                        new NewTopic(inputTopic, 1, replicationFactor),
+                        new NewTopic(outputTopic, 1, replicationFactor)));
+
+        producer.send(new ProducerRecord<>(inputTopic, 1));
+        producer.send(new ProducerRecord<>(inputTopic, 2));
+        producer.send(new ProducerRecord<>(inputTopic, 3));
+
+        // run the Flink job
+        FLINK.submitJob(
+                new JobSubmission.JobSubmissionBuilder(kafkaExampleJar)
+                        .setDetached(false)
+                        .addArgument("--input-topic", inputTopic)
+                        .addArgument("--output-topic", outputTopic)
+                        .addArgument("--prefix", "PREFIX")
+                        .addArgument(
+                                "--bootstrap.servers",
+                                String.join(
+                                        ",",
+                                        KAFKA_CONTAINER.getBootstrapServers(),
+                                        KAFKA_CONTAINER.getNetworkAliases().stream()
+                                                .map(
+                                                        host ->
+                                                                String.join(
+                                                                        ":",
+                                                                        host,
+                                                                        Integer.toString(9092)))
+                                                .collect(Collectors.joining(","))))
+                        .addArgument("--group.id", "myconsumer")
+                        .addArgument("--auto.offset.reset", "earliest")
+                        .addArgument("--transaction.timeout.ms", "900000")
+                        .addArgument("--flink.partition-discovery.interval-millis", "1000")
+                        .build());
+        final Properties consumerProperties = new Properties();
+        consumerProperties.put(
+                CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG,
+                KAFKA_CONTAINER.getBootstrapServers());
+        consumerProperties.put(
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, VoidDeserializer.class);
+        consumerProperties.put(
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class);
+        List<Integer> records =
+                KafkaUtil.drainAllRecordsFromTopic(outputTopic, consumerProperties).stream()
+                        .map(r -> ByteBuffer.wrap(r.value()).getInt())
+                        .collect(Collectors.toList());
+        assertThat(records).hasSize(3).containsExactly(1, 2, 3);
+    }
```

Comment:
I think only resubmission does not really change much for the connector. Maybe a simple recovery we can add in the future a simple recovery smoke test although the source/sink testing framework definitely covers these cases.
So far in my opinion the smoke tests for connectors should catch the basic class not found exceptions and serialization problems.

Reply:
Smoke test should only focus on the happy path. It might make sense to define the scope of the scope of critical functionalities(happy path only) as the next step. Problems like Class not found exceptions and serialization problems are the background tech issues that break these critical functionalities.